File: backend/semgrep_rules/tests/python/kisa_sw_28_improper_certificate_validation.py
import requests
def unsafe():
    # ruleid: kisa.sw28.python.certificate-validation-disabled
    return requests.get("https://example.com", verify=False)
def safe():
    # ok: kisa.sw28.python.certificate-validation-disabled
    return requests.get("https://example.com", verify=True)

File: backend/semgrep_rules/tests/javascript/kisa_sw_28_improper_certificate_validation.js
const https = require("https");
function unsafe() {
  // ruleid: kisa.sw28.javascript.certificate-validation-disabled
  return new https.Agent({ rejectUnauthorized: false });
}
function safe() {
  // ok: kisa.sw28.javascript.certificate-validation-disabled
  return new https.Agent({ rejectUnauthorized: true });
}
